src/Illuminate/Database/Query/Builder.php

```diff
@@ -278,6 +278,11 @@ public function addSelect($column)
 	{
 		$column = is_array($column) ? $column : func_get_args();
 
+		if (is_null($this->columns))
+		{
+			$this->columns = [ '*' ];
+		}
+
 		$this->columns = array_merge((array) $this->columns, $column);
```

Comment:
No this is fine then. I didn't consider empty arrays as being useful. We could probably ditch the array cast in `array_merge` though, as we should now always have an array.

Do please remove the surrounding whitespace in `[ '*' ]` though :wink: